common/utils_unittest.cpp

```diff
@@ -111,33 +112,48 @@ UTILS_TEST(guessResolutionBasic) {
     EXPECT_EQ(h, 0);
 }
 
-UTILS_TEST(guessResolutionOverflow) {
-    std::stringstream sstream;
-    int w, h;
-
-    w = h = std::numeric_limits<int>::max();
-
-    // sanity check
-    ASSERT_GT(std::numeric_limits<long>::max(), w);
-    ASSERT_GT(std::numeric_limits<long>::max(), h);
-
-    sstream << std::numeric_limits<long>::max()
-            << "x"
-            << std::numeric_limits<long>::max();
-
-    EXPECT_FALSE(guessResolution(sstream.str().c_str(), w, h));
-    EXPECT_EQ(w, 0);
-    EXPECT_EQ(h, 0);
-}
-
-UTILS_TEST(getPlaneResolutionDeath) {
-    ::testing::FLAGS_gtest_death_test_style = "threadsafe";
+struct BppEntry {
+    uint32_t fourcc;
     uint32_t planes;
-    uint32_t w[3], h[3];
-
-    // getPlaneResolution aborts program (via assert) if there is an invalid
-    // fourcc value
-    EXPECT_DEATH(
-        getPlaneResolution(0, 1920, 1080, w, h, planes),
-        "do not support this format");
+    float bpp;
+};
+
+const static BppEntry bppEntrys[] = {
+    { VA_FOURCC_I420, 3, 1.5 },
+    { VA_FOURCC_YV12, 3, 1.5 },
+    { VA_FOURCC_IMC3, 3, 1.5 },
+    { VA_FOURCC_422H, 3, 2 },
+    { VA_FOURCC_422V, 3, 2 },
+    { VA_FOURCC_444P, 3, 3 },
+    { VA_FOURCC_YUY2, 1, 2 },
+    { VA_FOURCC_UYVY, 1, 2 },
+    { VA_FOURCC_RGBX, 1, 4 },
+    { VA_FOURCC_RGBA, 1, 4 },
+    { VA_FOURCC_BGRX, 1, 4 },
+    { VA_FOURCC_BGRA, 1, 4 },
+};
+
+UTILS_TEST(getPlaneResolution)
+{
+
+    uint32_t width = 1920;
+    uint32_t height = 1080;
+    uint32_t planes;
+    uint32_t w[3];
+    uint32_t h[3];
+    EXPECT_FALSE(getPlaneResolution(0, 1920, 1080, w, h, planes));
+    EXPECT_EQ(planes, 0);
```

Comment:
This breaks compilation for sign comparison warning.  Also, the expected value should be the first parameter (e.g. `EXPECT_EQ(0u, planes)`)